node-ui/src/hooks/useInternetComputer.tsx

```diff
@@ -0,0 +1,352 @@
+
+        // console.log('currentNetwork:', currentNetwork);
+        // return;
+        if (walletSignatureData) {
+          if (walletSignatureData.payload?.message.message) {
+            const message = new Uint8Array(
+              Buffer.from(walletSignatureData.payload?.message.message),
+            );
+            const encodedChallenge = new Uint8Array(Buffer.from(message));
+            console.log('Current Network:', currentNetwork);
+            let delegationIdentity;
+            try {
+              delegationIdentity = await authWithII({
+                url: currentNetwork.url,
+                sessionPublicKey: encodedChallenge,
+              });
+              console.log('delegationIdentity:', delegationIdentity);
+            } catch (error: any) {
+              console.error('Error:', error);
+              setErrorMessage(
+                error.message || 'An error occurred during authentication.',
+              );
+              setReady(true);
+              return; // Exit the function if there's an error
+            }
+
+            if (delegationIdentity) {
+              const data = { encodedChallenge, delegationIdentity };
+              // Serialize the data with BigInt and Uint8Array handling
+              const serializedData = JSON.stringify(data, (_, v) => {
+                if (typeof v === 'bigint') {
+                  // Convert BigInt to hex string
+                  return v.toString(16);
+                }
+                if (v instanceof Uint8Array) {
+                  // Convert Uint8Array to hex string
+                  return uint8ArrayToHexString(v);
+                }
+                return v;
+              });
+              const jsonData = JSON.parse(serializedData);
+              const delegationChain = {
+                delegations: jsonData.delegationIdentity.delegations,
+                publicKey: jsonData.delegationIdentity.userPublicKey,
+              };
+              const publicKey: string = Buffer.from(
+                jsonData.delegationIdentity.userPublicKey,
+              ).toString('base64');
+
+              const walletMetadata: WalletMetadata = {
+                wallet: WalletType.INTERNETCOMPUTER({
+                  cannisterId: currentNetwork.cannisterId,
+                  walletName: 'Internet Identity',
+                }),
+                verifyingKey: publicKey,
+              };
+              if (walletSignatureData?.payload) {
+                const internetComputerRequest: LoginRequest = {
+                  walletSignature: JSON.stringify(delegationChain),
+                  payload: walletSignatureData.payload,
+                  walletMetadata: walletMetadata,
+                };
+
+                console.log(
+                  'internetComputerRequest:',
+                  internetComputerRequest,
+                );
+
+                const result: ResponseData<LoginResponse> = isLogin
```

Comment:
it is strange that both call return the same type

Reply:
They are similar calls with the exception of checking for root keys and storing client keys in login() request.